### PR TITLE
Voucher server logging improvements

### PR DESCRIPTION
**Pull request by jacobmichels**
This PR does two things:
1. Modifies every `check` struct to include a `voucher.CheckLogger`, an interface that abstracts over a logger. The goal is to provide a consistent logging experience across all voucher checks.
2. Modifies the `LogInfo`, `LogWarning`, and `LogError` functions defined in server/log.go to:
  a. accept caller file and caller function strings to pinpoint where the log happened for debugging
  b. use structured logging with logrus fields so the log format is more predictable

## Related
addresses https://github.com/Shopify/voucher-internal/issues/198

### CODE REVIEW

**Comment by jacobmichels** [comment]
@clintonShopify What do you think about this approach? I'm open to suggestions 😃

**Comment by rxbchen** [comment]
Overall I really like this change.  By doing this, we can easy switch loggers in the future (as logrus is currently not maintained)

File: v2/check_logger.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package voucher
+
+type CheckLogger interface {
+	LogDebug(check, callerFile, callerFunction, msg string)
```
Comment (thepwagner) [inline v2/check_logger.go]: Manually specifying `check`, `callerFile`, and `callerFunction` smells a little strange to me.
You could simplify this interface by using `runtime.Caller()` to look these values up - https://go.dev/play/p/VRKdukRDFGw

If you keep pulling at this thread, `logrus.FieldLogger` could be an alternative to the custom `CheckLogger` interface.

**Review by thepwagner — COMMENTED** [review]
I like the idea of passing checks a logger reference instead of relying on the global/singleton `logrus.StandardLogger()` instance.

The code to set `(check.logger)` seems legit, but I don't see how any of the store logger references are actually used - we're still using `LogWarning` and `LogError` to emit messages, and those don't seem to care about the stored `CheckLogger` that was set on the check.
If we want to use the static logger, could we roll back the other stuff?
If we don't want to use the static logger, could we finish migrating to the logger-by-reference?

--- 

Inline I included a snippet for determining the calling file/function at runtime, instead of relying on future-us to maintain these strings.

FWIW Logrus also has this functionality built in, the goal of this PR could also be met by:
* Adding `logrus.SetReportCaller(true)` to an `init()` or `main()` function
* Replacing the `LogError()` function calls with direct references to `logrus.Error()`

Doing the former without the latter will include the call site information in the logs, but it would look like all messages originated from `LogError`, not the function that called `LogError`.

Behind the scenes it's doing the same thing as the golang playground: [capturing the stack and walking backwards](https://github.com/sirupsen/logrus/blob/85981c045988a3ebe607e2f5b8a6499d86e0ddac/entry.go#L179-L213).
Here's a second [logrus-focused playground](https://go.dev/play/p/-UORRXtZpG4)

**Comment by jacobmichels** [comment]
I thought about this for a while, and decided to roll back the logging changes to the checks. It doesn't make sense to be using the static logger for some things and a reference to a logger for others. None of the checks actually need a logger at the moment so it's probably better for them to just not have one.

I did modify server/log.go to print out the caller file+line and function as fields in log messages. I think this could make it easier to debug where messages are coming from, but it does make the logs pretty heavy when running locally. An example log is: `{"file":"/Users/jacobmichels/src/github.com/Shopify/voucher/v2/server/server.go:38","func":"github.com/grafeas/voucher/v2/server.(*Server).SetCheckGroup","level":"info",...}`

**Comment by jacobmichels** [comment]
Closing this PR, I don't think my proposed changes add any value.

**Pull request closed without merging**